numpy/core/src/common/npy_partition.h

```diff
@@ -0,0 +1,27 @@
+#ifndef __NPY_PARTITION_H__
+#define __NPY_PARTITION_H__
+
+#include "npy_sort.h"
+
+/* Python include is for future object sorts */
+#include <Python.h>
+
+#include <numpy/ndarraytypes.h>
+#include <numpy/npy_common.h>
+
+#define NPY_MAX_PIVOT_STACK 50
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+NPY_NO_EXPORT PyArray_PartitionFunc *
```

Comment:
These get functions are the C level interface to the outer world. They were already existing prior to that commit, I just moved the logic to the C++ file to avoid cluttering the C header with C++ stuff. But in the end we get the same interface as before.